### ISSUE TITLE
Option to always choose stack interactively

### ISSUE THREAD

**Issue by pbzdyl** [pulumi/pulumi#2312]
It would be great to have an option to make pulumi command to always ask which stack should be used. I know I can use `pulumi preview` or `pulumi up` with `--stack <stack name>` but for interactive sessions in terminal it would be helpful to have these commands to ask to select the stack in the same way `pulumi stack select` does.

**Comment by pbzdyl** [comment]
I don't personally like having a default context to be set implicitly in background (e.g. for `kubectl` or for `pulumi`) when working interactively in terminal - making the choice of the context for the command being run explicit if much more safe in my opinion. I know I can create e.g. shell aliases with added `--stack ...` arguments for different stacks but it has additional friction. Also I often run the command (e.g. `pulumi up` or `pulumi preview`) and once the preview is displayed I noticed I have the incorrect stack selected and I have to change the stack and rerun the command.

**Comment by pbzdyl** [comment]
And I just found #1831 and for the feature I'm suggesting I wouldn't like the selection to be sticky :)

**Comment by keen99** [comment]
agreed for the same reasons.

additionally, since the stack selection is stored...somewhere that's outside of the shell's environment... I can't sanely work on two stacks in parallel. (it works, but it feels wrong. and the last one to return seems to be the one that sets the next current default..)   (I'm assuming the currect stack selection is stored somewhere in/around state, upstream, based on the time it takes select and ls to return...).   

I wouldn't mind the ability to set/select a stack as an environment variable, either.

**Comment by Mondonno** [comment]
Agree, it would be very useful to have a pulumi option to enable stack choosing with every command involving it that wouldn't be sticky.
